components/hab/src/main.rs

```diff
@@ -1669,8 +1666,7 @@ fn sub_ring_key_import(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
 
 fn sub_service_key_generate(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     let org = org_param_or_env(m)?;
-    let service_group = required_value_of(m, "SERVICE_GROUP");
-    let service_group = ServiceGroup::from_str(service_group)?;
+    let service_group = required_value_of(m, "SERVICE_GROUP").parse::<ServiceGroup>()?;
```

Comment:
Unlike the two instances above, with this call, you don't have to supply a type hint (using the `::<>` "turbofish" operator). The compiler has enough information here to know that `parse` is expected to return a `ServiceGroup`. That's because `command::service::key::generate::start` takes a `&ServiceGroup` as an argument. In the other calls, a service group was only being used to create a String, without any intermediate explicit typing that would indicate that we should have parsed to a `ServiceGroup`.

`parse` is one of the common times you'll need to use the `::<>` operator, but if it's not required, it's generally considered good form to leave it off and let the compiler do the work for you.